stdlib/LinearAlgebra/src/triangular.jl

```diff
@@ -2526,6 +2551,66 @@ Base.@propagate_inbounds function _sqrt_quasitriu_offdiag_block_2x2!(R, A, i, j)
     return R
 end
 
+# solve Sylvester's equation AX + XB = -C using blockwise recursion until the dimension of
+# A and B are no greater than blockwidth, based on Algorithm 1 from
+# Jonsson I, Kågström B. Recursive blocked algorithms for solving triangular systems—
+# Part I: one-sided and coupled Sylvester-type matrix equations. (2002) ACM Trans Math Softw.
+# 28(4).
+function _sylvester_quasitriu!(A, B, C; blockwidth=64, nA=checksquare(A), nB=checksquare(B), catcherr=false)
+    kwargs = (blockwidth=blockwidth, catcherr=catcherr)
+    if 1 ≤ nA ≤ blockwidth && 1 ≤ nB ≤ blockwidth  # base case
+        try
+            _, scale = LAPACK.trsyl!('N', 'N', A, B, C)
+            rmul!(C, -inv(scale))
+        catch e
+            if !(catcherr && e isa LAPACKException)
+                throw(e)
+            end
+        end
```

Comment:
For just about any large real triangular matrix `A` created as `A=UpperTriangular(randn(n, n))^2`, `sqrt(A)^2` fails to be approximately equal to `A` (the top right terms explode). This is true for the point algorithm in 1.6 and on master as well. In the blocked version, `LAPACK.trsyl!` will throw a `LAPACKException` for these matrices, I guess indicating that it could not solve Sylvester's equation. But if that exception is caught, the result seems to be equivalent to what the point algorithm would have returned. Hence why we have this `try`/`catch`.

Reply:
maybe a comment to justify ignoring LAPACKException? Should it get a warning when convergence fails?

Reply:
I'll add a comment. RE a warning, perhaps? Not really certain what the conventions are in the stdlib regarding warnings.

Reply:
While writing the tests for `triangular.jl` I realized how poorly conditioned `UpperTriangular(randn(n, n))` is. It's an easy way to generate a triangular test matrix but it's often not representative for real world triangular matrices that are typically a result of a matrix factorization. E.g.
```julia
julia> mean(cond(lu(randn(100, 100)).U) for i in 1:10000)
2309.908474551785

julia> mean(cond(triu(randn(100, 100))) for i in 1:10000)
2.915729043528958e19
```
so I generally think it would be better to construct triangular test matrices from an LU.

A separate question is the handling of non-zero `info` values in LAPACK (or almost any other mathematical function defined in C/Fortran). I'm not sure we made the right decision when we made all of these throw instead of returning the exit status in some form. However, that is a very big issue to tackle so the your current solution is probably fine (although I generally find `try/catch` pretty crude).